python/python/ert_gui/simulation/run_dialog.py

```diff
@@ -222,3 +232,48 @@ def hideKillAndShowDone(self):
         self.processing_animation.hide()
         self.kill_button.setHidden(True)
         self.done_button.setHidden(False)
+        self.realizations_view.setVisible(True)
+        self.restart_button.setVisible(self.has_failed_realizations())
+
+    def has_failed_realizations(self):
+        completed = self._run_model._completed_realizations_mask
+        initial = self._run_model._initial_realizations_mask
+        for (index, successful) in enumerate(completed):
+            if initial[index] and not successful:
+                return True
+        return False
+
+    def count_successful_realizations(self):
+        completed = self._run_model._completed_realizations_mask
+        initial = self._run_model._initial_realizations_mask
+        sum = 0
+        for (index, successful) in enumerate(completed):
+            if initial[index] and successful:
```

Comment:
Is it necessary to iterate over the initial realizations? And should a realization that was `initial[index]` `False` even be restarted?